### PR TITLE
✨ Implement custom errors 2.0

### LINKED ISSUE #8079: Custom error templates 2.0
For some background to this issue, you can read my comment on #6894. 

Currently we have support for a custom error template. Theme developers can add an `error.hbs` template and this will be used for rendering all errors.

We would like to upgrade this behaviour to the following:

`error.hbs` - will stay as a fallback, and continue to work.
Providing `error-xxx.hbs` will allow theme developers to target specific errors or classes of errors.

E.g. 

- `error-404.hbs` would be used to render any 404 error.
- `error-4xx.hbs` would be used to render any other 4xx error
- `error-5xx.hbs` would be used to render any 5xx error
- `error.hbs` would be used if there is not a more specific error template.

The most common setup, I imagine will be providing both `error.hbs` and `error-404.hbs`.

However this cascading setup will provide maximum control & futureproof us against situations where we have all kinds of different errors exposed to themes from more advanced functionality.

### PR DESCRIPTION
**Pull request by ErisDS**
This PR builds on the work in #8147 to implement the cascading custom error template setup that was spec'd in #8079. It is only WIP because #8147 needs to be merged first.

There are a few additional changes:

- the addition of `defaultViews` in config/overrides.json because the error template is not anything to do with the admin panel! I also want to move default.hbs, which IS for the admin panel to core/server/admin/views as noted in TryGhost/Ghost-Admin#574
- I renamed the error template from user-error.hbs to error.hbs, as it's not only shown for user errors and that is a bit odd.

closes #8079 

- reimplement custom errors, but with a stack like channels & single templates
- change ghost_head to only not output on 500+ server errors, rather than 400+ user errors
- add coverage for the new template functions

### CODE REVIEW

**Comment by ErisDS** [comment]
This one is ready 😁

File: core/server/config/overrides.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
         "clientAssets": "core/built/assets",
         "helperTemplates": "core/server/helpers/tpl/",
         "adminViews": "core/server/views/",
+        "defaultViews": "core/server/views/",
```
Comment (kirrg001) [inline core/server/config/overrides.json]: adminViews === defaultViews?on purpose?

Comment (ErisDS) [inline core/server/config/overrides.json]: Yes, they are not the same thing - adminViews are for the admin panel, defaultViews are fallbacks for themes. They live in the same place for noooow. Admin views will change here: https://github.com/TryGhost/Ghost/pull/8142/files#diff-4a9ff835bae8d805a7741e6356728bcfR5

Comment (kirrg001) [inline core/server/config/overrides.json]: great thanks 👍

**Review by kirrg001 — APPROVED** [review]
LGTM 👍 
Will merge if master is green. Somehow the last commit on master failed a lot. I saw some yarn errors.

**Pull request merged**